wordle-web/scripts/wordService.ts

```diff
@@ -0,0 +1,41 @@
+import { LetterState } from "./letter";
+import { WordList } from "./wordList";
+
+export class WordService {
+  static validGuessedWords(guesses: string[], stateArray: LetterState[][]): string[] {
+    const validWords = new Set<string>();
+
+    for(let j = 0; j < guesses.length; j++){
+      const currentGuess: string = guesses[j];
+      const states: LetterState[] =  stateArray[j];
+
+      const isValidWord = WordList.filter(word => {
+        let isValid = true;
+
+        for(let i = 0; i < word.length; i++) {
+          const guessedChar = (i < currentGuess.length) ? currentGuess[i].toLowerCase() : null;
```

Comment:
```suggestion
          const guessedChar = currentGuess[i].toLowerCase();
```
This will always be true so you can remove it. There won't be a case where the word length is longer than the guessed word length.

Reply:
Changed this completely.